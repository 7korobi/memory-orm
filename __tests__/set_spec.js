/*
 * decaffeinate suggestions:
 * DS102: Remove unnecessary code created because of implicit returns
 * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
 */
require('./models/index')
const Mem = require('../lib/index')

const keys = [
  'tag',
  'face',
  'chr_set',
  'chr_npc',
  'chr_job',
  'locale',
  'random',
  'folder',
  'role',
  'able',
  'sow_village',
  'sow_village_plan',
  'sow_roletable',
]

keys.map((key) =>
  describe(key, function () {
    test('all is query', () => expect(Mem.Set[key].all).toEqual(Mem.Query[key + 's']))
    test('$name snapshot', () => expect(Mem.Set[key].$name).toMatchSnapshot())
  })
)
