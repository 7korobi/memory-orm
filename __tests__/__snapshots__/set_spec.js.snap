// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`able $name snapshot 1`] = `
Object {
  "base": "able",
  "depends": Array [
    "ables",
  ],
  "deploys": Array [],
  "id": "able_id",
  "ids": "able_ids",
  "list": "ables",
  "relations": Array [
    "roles",
  ],
}
`;

exports[`chr_job $name snapshot 1`] = `
Object {
  "base": "chr_job",
  "depends": Array [
    "chr_jobs",
  ],
  "deploys": Array [
    [Function],
  ],
  "id": "chr_job_id",
  "ids": "chr_job_ids",
  "list": "chr_jobs",
  "relations": Array [
    "chr_set",
    "face",
  ],
}
`;

exports[`chr_npc $name snapshot 1`] = `
Object {
  "base": "chr_npc",
  "depends": Array [
    "chr_npcs",
  ],
  "deploys": Array [
    [Function],
  ],
  "id": "chr_npc_id",
  "ids": "chr_npc_ids",
  "list": "chr_npcs",
  "relations": Array [
    "chr_set",
    "chr_job",
    "face",
  ],
}
`;

exports[`chr_set $name snapshot 1`] = `
Object {
  "base": "chr_set",
  "depends": Array [
    "chr_sets",
  ],
  "deploys": Array [],
  "id": "chr_set_id",
  "ids": "chr_set_ids",
  "list": "chr_sets",
  "relations": Array [
    "chr_jobs",
    "chr_npcs",
  ],
}
`;

exports[`face $name snapshot 1`] = `
Object {
  "base": "face",
  "depends": Array [
    "faces",
  ],
  "deploys": Array [
    [Function],
  ],
  "id": "face_id",
  "ids": "face_ids",
  "list": "faces",
  "relations": Array [
    "tags",
    "chr_jobs",
    "chr_npcs",
  ],
}
`;

exports[`folder $name snapshot 1`] = `
Object {
  "base": "folder",
  "depends": Array [
    "folders",
  ],
  "deploys": Array [
    [Function],
  ],
  "id": "folder_id",
  "ids": "folder_ids",
  "list": "folders",
  "relations": Array [],
}
`;

exports[`locale $name snapshot 1`] = `
Object {
  "base": "locale",
  "depends": Array [
    "locales",
  ],
  "deploys": Array [],
  "id": "locale_id",
  "ids": "locale_ids",
  "list": "locales",
  "relations": Array [],
}
`;

exports[`random $name snapshot 1`] = `
Object {
  "base": "random",
  "depends": Array [
    "randoms",
  ],
  "deploys": Array [
    [Function],
  ],
  "id": "random_id",
  "ids": "random_ids",
  "list": "randoms",
  "relations": Array [],
}
`;

exports[`role $name snapshot 1`] = `
Object {
  "base": "role",
  "depends": Array [
    "roles",
  ],
  "deploys": Array [],
  "id": "role_id",
  "ids": "role_ids",
  "list": "roles",
  "relations": Array [
    "ables",
  ],
}
`;

exports[`sow_roletable $name snapshot 1`] = `
Object {
  "base": "sow_roletable",
  "depends": Array [
    "sow_roletables",
  ],
  "deploys": Array [],
  "id": "sow_roletable_id",
  "ids": "sow_roletable_ids",
  "list": "sow_roletables",
  "relations": Array [],
}
`;

exports[`sow_village $name snapshot 1`] = `
Object {
  "base": "sow_village",
  "depends": Array [
    "sow_villages",
  ],
  "deploys": Array [
    [Function],
  ],
  "id": "sow_village_id",
  "ids": "sow_village_ids",
  "list": "sow_villages",
  "relations": Array [
    "turns",
    "marks",
    "option_datas",
    "folder",
    "say",
    "mob",
    "game",
  ],
}
`;

exports[`sow_village_plan $name snapshot 1`] = `
Object {
  "base": "sow_village_plan",
  "depends": Array [
    "sow_village_plans",
  ],
  "deploys": Array [],
  "id": "sow_village_plan_id",
  "ids": "sow_village_plan_ids",
  "list": "sow_village_plans",
  "relations": Array [],
}
`;

exports[`tag $name snapshot 1`] = `
Object {
  "base": "tag",
  "depends": Array [
    "tags",
  ],
  "deploys": Array [],
  "id": "tag_id",
  "ids": "tag_ids",
  "list": "tags",
  "relations": Array [
    "chr_set",
    "faces",
    "nodes",
    "tag",
  ],
}
`;
